src/server/game/AI/ScriptedAI/ScriptedCreature.h

```diff
@@ -210,11 +210,16 @@ struct TC_GAME_API ScriptedAI : public CreatureAI
     //Plays a sound to all nearby players
     void DoPlaySoundToSet(WorldObject* source, uint32 soundId);
 
-    //Drops all threat to 0%. Does not remove players from the threat list
-    void DoResetThreat();
-
-    float DoGetThreat(Unit* unit);
-    void DoModifyThreatPercent(Unit* unit, int32 pct);
+    // Add specified amount of threat directly to victim (ignores redirection effects) - also puts victim in combat and engages them if necessary
+    void AddThreat(Unit* victim, float amount, Unit* who = nullptr) { (who ? who : me)->AddThreat(victim, amount, nullptr, true, true); }
```

Comment:
The general idea is to make each header only include things it needs for itself - all kind of convenience helpers don't belong in this category and should be moved to .cpp file